src/grid/Grid.tsx

```diff
@@ -91,6 +91,28 @@ export const Slot: FC<SlotProps> = ({ tile, style, className, ...props }) => (
   />
 );
 
+interface Offset {
+  x: number;
+  y: number;
+}
+
+/**
+ * Gets the offset of one element relative to an ancestor.
+ */
+function offset(element: HTMLElement, relativeTo: Element): Offset {
+  if (
+    !(element.offsetParent instanceof HTMLElement) ||
+    element.offsetParent === relativeTo
+  ) {
+    return { x: element.offsetLeft, y: element.offsetTop };
+  } else {
+    const o = offset(element.offsetParent, relativeTo);
+    o.x += element.offsetLeft;
+    o.y += element.offsetTop;
+    return o;
+  }
+}
```

Comment:
Checking the client rect in combination with the body's scroll offset is a nice idea. I think to have the same correctness guarantees as this function though, you would have to traverse the DOM and subtract the scroll offsets for each scroll container that is a child of the DOM, as well. So in the end you'd still need a recursive process similar to this one. Unless I'm missing something?

I would note that in practice, `relativeTo` will only be 1-3 levels above the `element`, and I do think we want to account for non-body scroll containers in case we make some changes to how the call layouts are layered.

Reply:
3 levels should be totally fine then. 
The idea was that the scroll offset is the same. for all elements (but that is most likely incorrect if we have multiple scroll containers)
Then the scroll offset could be ignored. (it would get substracted away)